File: API/Models/DTOs/StudentDTO.cs
using System;

namespace API.Models.DTOs;

public class StudentDTO
{
    public required string FirstName { get; set; }
    public string? SecondName { get; set; }
    public required string LastName { get; set; }
    public required string IdentityNumber { get; set; }
    public required string Gender { get; set; }
    public string? Email { get; set; }
    public string? Phone { get; set; }
    public required DateOnly DateOfBirth { get; set; }
    public required AddressDTO AddressDTO { get; set; }
    // public required GuardianDTO GuardianDTO { get; set; }
}

File: Client/src/app/_interfaces/student-dto.ts
import { AddressDto } from "./address-dto";

export interface StudentDto {
      firstName: string;
      secondName?: string;
      lastName: string;
      gender: string;
      identityNumber: string;
      email?: string;
      phone?: string;
      dateOfBirth: string;
      AddressDto: AddressDto;
}

